src/BookSystem/Interfaces/IBookRepository.cs

```diff
@@ -0,0 +1,9 @@
+namespace BookSystem;
+public interface IBookRepository
+{
+    void AddBook(Book book);
+    void UpdateBook(string title, string? newTitle = null, string? newAuthor = null, string? newGenre = null, DateTime? newPublicationDate = null);
```

Comment:
It won't be better to update the model instead of passing many parameters

Reply:
These parameters are required to update, but you can also implement another class to do the same job and just return the bookUpdateModel.

    public class BookUpdateModel
    {
        public string? NewTitle { get; set; }
        public string? NewAuthor { get; set; }
        public string? NewGenre { get; set; }
        public DateTime? NewPublicationDate { get; set; }
    }

Reply:
Okay!
I meant another thing, but we will see later.